Makefile

```diff
@@ -1,6 +1,7 @@
 # simple makefile to simplify repetitive build env management tasks under posix
 
 PYTHON ?= python
+DEFAULT_MESON_BUILD_DIR = build/cp$(shell python -c 'import sys; print(f"{sys.version_info.major}{sys.version_info.minor}")' )
```

Comment:
why not just remove ``build/`` ? It's not like the build takes hours.

Reply:
In principle you can have different build folders under `build` and this does not make too much sense to delete them all.

Personally, these days I have many different scikit-learn environments that I create from lock-file to try to reproduce CI issues (especially this week :sweat:) so I would not want to delete them all. Right now I have 3 folders `build/cp39`, `build/cp311`, `build/cp312`.

Reply:
In your setup would you get weird error messages in the other envs if you deleted `build/` or would it "just" rebuild?

For me it is fine to only delete one sub-directory, but I'd also be happy to just remove all of `build/`

Reply:
You get error message like this if you delete your build folder:
```
❯ python -c 'import sklearn'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1322, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1262, in _find_spec
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/site-packages/_scikit_learn_editable_loader.py", line 310, in find_spec
    tree = self._rebuild()
           ^^^^^^^^^^^^^^^
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/site-packages/_scikit_learn_editable_loader.py", line 338, in _rebuild
    if self._work_to_do(env):
       ^^^^^^^^^^^^^^^^^^^^^
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/site-packages/_scikit_learn_editable_loader.py", line 325, in _work_to_do
    p = subprocess.run(dry_run_build_cmd, cwd=self._build_path, env=env, capture_output=True)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/lesteve/micromamba/envs/testenv/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/lesteve/scikit-learn/build/cp312'
```

So this would go like this:
- env1 is broken let's `rm -rf build` and do `make dev` to rebuild from scratch
- OK let's switch to env2 and now work on this other thing
- ah OK weird error in env2 since I `rm -rf build` ah I did not think of this :sweat: 
- env2 needs to be rebuilt too oh well one of this day ...